meson.build

```diff
@@ -0,0 +1,116 @@
+project('dosbox-staging', 'c', 'cpp',
+        default_options : ['cpp_std=c++14', 'b_ndebug=if-release'],
+        version : '0.77.0')
+
+
+# compiler flags
+#
+cc = meson.get_compiler('c')
+cxx = meson.get_compiler('cpp')
+
+add_global_arguments('-Weffc++', language : 'cpp')
+if cxx.has_argument('-Wextra-semi')
+    add_global_arguments('-Wextra-semi', language : 'cpp')
+endif
+
+
+# config file
+#
+conf_data = configuration_data()
+conf_data.set('version', meson.project_version())
+conf_data.set10('C_MODEM', get_option('use_sdl2_net'))
+conf_data.set10('C_IPX', get_option('use_sdl2_net'))
+conf_data.set10('C_OPENGL', get_option('use_opengl'))
+conf_data.set10('C_FLUIDSYNTH', get_option('use_fluidsynth'))
+conf_data.set10('C_SSHOT', get_option('use_png'))
+
+if cc.has_function('clock_gettime', prefix : '#include <time.h>')
+  conf_data.set('DB_HAVE_CLOCK_GETTIME', true)
+endif
+
+if cc.has_member('struct dirent', 'd_type', prefix : '#include<dirent.h>')
+  conf_data.set('DIRENT_HAS_D_TYPE', true)
+endif
+
+configure_file(input : 'src/config.h.in', output : 'config.h',
+               configuration : conf_data)
+
+
+# external dependencies
+#
+msg = 'You can disable this dependency with: -D@0@=false'
+dummy_dep = dependency('', required : false)
+threads_dep = dependency('threads')
+sdl2_dep = dependency('sdl2', version : '>= 2.0.2')
+
+if get_option('use_sdl2_net')
+  sdl2_net_dep = dependency('SDL2_net', version : '>= 2.0.1',
+                            not_found_message : msg.format('use_sdl2_net'))
```

Comment:
Wow.. could this get any cleaner? 
Ugh..

![2020-12-15_17-42](https://user-images.githubusercontent.com/1557255/102293957-fea77480-3efc-11eb-9796-c35fff774a7c.png)